flaml/autogen/agentchat/responsive_agent.py

```diff
@@ -268,11 +270,16 @@ def _append_oai_message(self, message: Union[Dict, str], role, conversation_id:
         """
         message = self._message_to_dict(message)
         # create oai message to be appended to the oai conversation that can be passed to oai directly.
-        oai_message = {k: message[k] for k in ("content", "function_call", "name", "role") if k in message}
+        oai_message = {k: message[k] for k in ("content", "function_call", "name", "role", "context") if k in message}
         # 'content' is a required field
-        if "content" not in oai_message or role not in ("assistant", "user", "function"):
+        if "content" not in oai_message:
+            return False
+        if role is not None:
+            oai_message["role"] = role
```

Comment:
This will still change "function" role to "user" and fail the function call feature.